### PR TITLE
SpatialConstAffine with test

### PR DESCRIPTION
**Pull request by szagoruyko**
does a*x + b on NCHW tensor, a and b have size C and expanded to be NCHW. Used to fix BatchNorm for training.

### CODE REVIEW

File: SpatialConstAffine.lua [hunk 1]
```diff
@@ -0,0 +1,39 @@
+local SpatialConstAffine, parent = torch.class('inn.SpatialConstAffine', 'nn.Module')
+
+function SpatialConstAffine:__init(nOutputPlane, inplace)
+  parent.__init(self)
+  self.a = torch.Tensor(1,nOutputPlane,1,1)
```
Comment (fmassa) [inline SpatialConstAffine.lua]: it would be great to have a less hidden way of initialising `a` and `b`.
Maybe a method that takes a `SpatialBatchNormalization` and extract the relevant fields and set `a` and `b` to point to them ?

Comment (szagoruyko) [inline SpatialConstAffine.lua]: feels out of place for this module

Comment (fmassa) [inline SpatialConstAffine.lua]: Well, I think something should be done in here, it's not a trainable module, and `a` and `b` are initialized with garbage. maybe a function to set `a` and `b`, or pass `a` and `b` in the constructor...
You know what, if you pass `a` and `b` as arguments, you can have a generic `ConstAffine`, in whichever dimension you want.

Comment (szagoruyko) [inline SpatialConstAffine.lua]: it's not garbage, it's identity now. see reset function

Comment (fmassa) [inline SpatialConstAffine.lua]: right, it's not garbage.
but I still think that with minimal changes this could be way more generic. But let's leave it like that for the moment.

**Comment by fmassa** [comment]
Just to mention that this function currently only works in batch mode. Not really a problem in most cases (as it's meant to replace BatchNorm).

**Comment by szagoruyko** [comment]
@fmassa I renamed it to `inn.ConstAffine` and added 2D support
